packages/preferences/src/browser/preferences-decorator.ts

```diff
@@ -64,4 +65,8 @@ export class PreferencesDecorator implements TreeDecorator {
     decorations(tree: Tree): MaybePromise<Map<string, TreeDecoration.Data>> {
         return this.preferencesDecorations;
     }
+    // tslint:disable-next-line:no-any
+    escapeInvisibleChars(prefValue: any): any {
+        return prefValue && typeof prefValue === 'string' ? prefValue.replace(/\n/g, '\\n').replace(/\r/g, '\\r') : prefValue;
```

Comment:
Would be great if you could move the conversion to one file and reuse it.

Reply:
Moved the escape/unescape functions to `core/src/common/strings`.